openmc/mgxs_library.py

```diff
@@ -0,0 +1,2577 @@
+            clone._nu_fission = copy.deepcopy(self._nu_fission, memo)
+            clone._prompt_nu_fission = \
+                copy.deepcopy(self._prompt_nu_fission, memo)
+            clone._delayed_nu_fission = \
+                copy.deepcopy(self._delayed_nu_fission, memo)
+            clone._kappa_fission = copy.deepcopy(self._kappa_fission, memo)
+            clone._chi = copy.deepcopy(self._chi, memo)
+            clone._chi_prompt = copy.deepcopy(self._chi_prompt, memo)
+            clone._chi_delayed = copy.deepcopy(self._chi_delayed, memo)
+            clone._beta = copy.deepcopy(self._beta, memo)
+            clone._decay_rate = copy.deepcopy(self._decay_rate, memo)
+            clone._inverse_velocity = \
+                copy.deepcopy(self._inverse_velocity, memo)
+            clone._xs_shapes = copy.deepcopy(self._xs_shapes, memo)
+
+            memo[id(self)] = clone
+
+            return clone
+
+        # If this object has been copied before, return the first copy made
+        else:
+            return existing
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def energy_groups(self):
+        return self._energy_groups
+
+    @property
+    def num_delayed_groups(self):
+        return self._num_delayed_groups
+
+    @property
+    def representation(self):
+        return self._representation
+
+    @property
+    def atomic_weight_ratio(self):
+        return self._atomic_weight_ratio
+
+    @property
+    def fissionable(self):
+        return self._fissionable
+
+    @property
+    def temperatures(self):
+        return self._temperatures
+
+    @property
+    def scatter_format(self):
+        return self._scatter_format
+
+    @property
+    def order(self):
+        return self._order
+
+    @property
+    def num_polar(self):
+        return self._num_polar
+
+    @property
+    def num_azimuthal(self):
+        return self._num_azimuthal
+
+    @property
+    def total(self):
+        return self._total
+
+    @property
+    def absorption(self):
+        return self._absorption
+
+    @property
+    def scatter_matrix(self):
+        return self._scatter_matrix
+
+    @property
+    def multiplicity_matrix(self):
+        return self._multiplicity_matrix
+
+    @property
+    def fission(self):
+        return self._fission
+
+    @property
+    def nu_fission(self):
+        return self._nu_fission
+
+    @property
+    def prompt_nu_fission(self):
+        return self._prompt_nu_fission
+
+    @property
+    def delayed_nu_fission(self):
+        return self._delayed_nu_fission
+
+    @property
+    def kappa_fission(self):
+        return self._kappa_fission
+
+    @property
+    def chi(self):
+        return self._chi
+
+    @property
+    def chi_prompt(self):
+        return self._chi_prompt
+
+    @property
+    def chi_delayed(self):
+        return self._chi_delayed
+
+    @property
+    def num_orders(self):
```

Comment:
Since `self._order` is set in the `__init__` method, it can safely be returned as the default return value.

```
if self._order is not None:
     if self._scatter_format in (None, 'legendre'):
           return self._order + 1

return self._order
```